### ISSUE TITLE
Python 3.10 deprecation warnings due to `distutils` import

### ISSUE THREAD

**Issue by The-Compiler** [PyCQA/astroid#1282]
### Steps to reproduce

- Run `python3.10 -Wall -c "import astroid"`

### Current behavior

```
.../astroid/interpreter/_import/spec.py:20: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  import distutils
.../astroid/modutils.py:51: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
  from distutils.sysconfig import get_python_lib  # pylint: disable=import-error
```

### Expected behavior

No deprecation warnings.

### `python -c "from astroid import __pkginfo__; print(__pkginfo__.version)"` output

2.9.1-dev0 (on 6691a9e97373b28e15a76b747f78c041f2064df4)

### Additional context

Might be related to #1103, but seems like a separate issue.

**Comment by Pierre-Sassoulas** [comment]
Doing #1103 would work, but upgrading setuptools min version to one that do not use distutils would probably work (?). I think we won't change anything before 3.12 is available in github actions.

**Comment by DanielNoord** [comment]
Closed with #1323
